karate-core/src/main/java/com/intuit/karate/driver/DevToolsDriver.java

```diff
@@ -392,6 +392,61 @@ public void setDimensions(Map<String, Object> map) {
                 .param("bounds", temp).send();
     }
 
+    /**
+     * This method takes the parameters needed to emulate a device.
+     *
+     * @param emulateDeviceParams
+     *        This object looks for the below keys needed to emulate a device.
+     *        By default it's configured to have width and height of a Iphone X mobile.
+     *
+     *        userAgent : It's needed to be passed in every request emulating the device agent
+     *        width : Width of the device emulating.
+     *        height : Height of the device emulating.
+     *
+     */
+    private void setEmulateDeviceInternal(Map<String, Object> emulateDeviceParams) {
+        if(emulateDeviceParams != null && !emulateDeviceParams.isEmpty()) {
+            if(emulateDeviceParams.containsKey("userAgent")) {
+                String userAgent = (String) emulateDeviceParams.get("userAgent");
+                logger.info("Setting userAgent={}", userAgent);
+                method("Network.setUserAgentOverride").param("userAgent", userAgent).send();
+            }
+
+            // Defaulting to Iphone x settings
+            Integer width = 375;
+            Integer height = 812;
+            Integer deviceScaleFactor = 1;
+
+            Object widthObj = emulateDeviceParams.get("width");
+            if(widthObj != null) {
+                width = (Integer) widthObj;
+            }
+
+            Object heightObj = emulateDeviceParams.get("height");
+            if(heightObj != null) {
+                height = (Integer) heightObj;
+            }
+
+            Object deviceScaleFactorObj = emulateDeviceParams.get("deviceScaleFactor");
+            if(deviceScaleFactorObj != null) {
+                deviceScaleFactor = (Integer) deviceScaleFactorObj;
+            }
+
+            logger.info("Setting deviceMetrics width={}, height={}, deviceScaleFactor={}", width, height, deviceScaleFactor);
+            method("Emulation.setDeviceMetricsOverride")
+                    .param("width", width)
+                    .param("height", height)
+                    .param("deviceScaleFactor", deviceScaleFactor)
+                    .param("mobile", true)
+                    .send();
+        }
+
+    }
+
+    public void setEmulateDevice(Map<String, Object> emulateDeviceParams) {
```

Comment:
a single method will do. let's not use a setter, and a method (not a big deal) but `driver.emulateDevice()` can be how it looks like in a script

Reply:
@ptrthomas  If make it as an action, then I cannot pass a map from feature file. It has to be 3 individual parameters. Are you suggesting the below method signature ?

emulateDevice(String width, String height, String userAgent) {

}

Reply:
@ptrthomas  Modifying it to the above signature removes lot of code. As we only have 3 parameters, please suggest if i can go with this signature. Thanks !!

Reply:
yeah, that's fine. in case in future it blows up we can add the `Map`

Reply:
> not sure if i can pass a map from feature file

why not ?

Reply:
Sounds good. I will resubmit the changes.